File: 1227-number-of-equivalent-domino-pairs/1227-number-of-equivalent-domino-pairs.java
class Solution {

    public int numEquivDominoPairs(int[][] dominoes) {
        int[] map = new int[100];
        int cnt = 0;
        for (int[] pair : dominoes) {
            int u = pair[0];
            int v = pair[1];
            
            int num = (u > v) ? v * 10 + u : u * 10 + v;
            cnt += map[num];

            map[num]++;
        }
        return cnt;
    }
}

File: 1227-number-of-equivalent-domino-pairs/1227-number-of-equivalent-domino-pairs.go
func numEquivDominoPairs(dominoes [][]int) int {
    hash := make([]int, 100)
    cnt := 0
    for _, pair := range dominoes {
        u, v := pair[0], pair[1]
        num := 0
        if u > v {
            num = v * 10 + u
        } else {
            num = u * 10 + v;
        }
        cnt += hash[num]

        hash[num]++
    }
    return cnt
}